src/timesync/timesyncd-manager.c

```diff
@@ -852,6 +855,11 @@ int manager_connect(Manager *m) {
                                 return 0;
                         }
 
+                        if (m->runtime_servers_changed) {
+                                m->runtime_servers_changed = false;
+                                restart = false;
+                        }
+
                         if (restart && !m->exhausted_servers && m->poll_interval_usec) {
```

Comment:
This is not your fault, but could you update the last condition to `m->poll_interval_usec > 0`?

Reply:
Added a PR now, but I'm curious: how can it be negative?